ui/src/components/Publication/Creation/MainText.tsx

```diff
@@ -143,6 +145,57 @@ const MainText: React.FC = (): React.ReactElement | null => {
         updateReferences
     } = Stores.usePublicationCreationStore();
 
+    const [editReferenceModalVisible, setEditReferenceModalVisible] = React.useState(false);
+    const [modalReferenceInfo, setModalReferenceInfo] = React.useState<Interfaces.Reference>({
+        id: '',
+        type: 'URL',
+        publicationId: '',
+        text: '',
+        location: null
+    });
+
+    const URLMatching = (referenceContent: string): [Interfaces.ReferenceType, string, string | null] => {
```

Comment:
I have also refactored this function in OCT-351. We will need to catch up for this because all 3 of us worked in this file 😄 and there will be lots of conflicts